### PR TITLE
Reverse Sequence

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

**Comment by Krovatkin** [comment]
@diyessi @mbrookhart I'm thinking do adding the autodiff logic in a separate PR, while it seems pretty simple, I'll need to figure out how to do bprop in TF and adapt my tests.

**Comment by diyessi** [comment]
The autodiff is just going to be a sequence reverse on the deltas.

File: src/ngraph/op/reverse_sequence.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        class ReverseSequence : public util::RequiresTensorViewArgs
+        {
+        public:
+            /// \brief Constructs an arcsin operation.
+            ///
+            /// \param arg Node that produces the input tensor.
+            ReverseSequence(const std::shared_ptr<Node> arg,
+                            size_t batch_axis,
+                            size_t seq_axis,
+                            const Shape& seq_lengths);
```
Comment (mbrookhart) [inline src/ngraph/op/reverse_sequence.hpp]: the sequence lengths needs to be a Node, we chatted about it in person.

Comment (diyessi) [reply src/ngraph/op/reverse_sequence.hpp]: And an attribute the specifies the axis that is lengthed, and the axis that indexes the node length.

**Comment by Krovatkin** [comment]
@mbrookhart not yet :-) i don't hook up this kernel to interpreter yet :-)

**Pull request merged**